### ISSUE TITLE
port to modern kernel (>=5.10)

### ISSUE THREAD

**Issue by ordex** [avafinger/s5k4ec#1]
Any interest in porting this driver to a more modern kernel?
It seems that the sunxi-vfe infrastructure is not there anymore and thus the s5k4ec driver can't rely on it.

I presume the driver should be converted to a more modern i2c one. I see the mainline kernel ships a driver called s5k4ecgx - have you tried to get it going?

What is lacking at the moment is a proper dts/dtb file which would configure the module properly without using the VFE/CSI subsystem. Is that possible at all?

**Comment by avafinger** [comment]
> I see the mainline kernel ships a driver called s5k4ecgx - have you tried to get it going?

No. I have not looked at the current state of the driver on the mainline. If i recall correctly someone ported (a kernel developer) and used the driver, which was implicit on his mail message, but i can't find who was the guy.

> What is lacking at the moment is a proper dts/dtb file which would configure the module properly without using the VFE/CSI subsystem. Is that possible at all?

I think it is possible, but you will only get a dark and greenish image without implementing the VFE/CSI portion in the libcamera.
Maybe is already there, if i have time i will try to find out more about the driver, but i am not motivated and mostly working with ROCKCHIP platform.

**Comment by ordex** [comment]
Thanks for your reply @avafinger !
Should you find that email, it may be a good starting point for who wants to give it another try.

In any case, good luck with the rockchip!
